coffeeshop/src/main/java/ro/eduarddudu/coffeeshop/ShopFinder.kt

```diff
@@ -0,0 +1,61 @@
+package ro.eduarddudu.coffeeshop
+
+import android.location.Location
+import ro.eduarddudu.coffeeshop.Objects.Shop
+import java.util.*
+
+object ShopFinder {
+
+    @Throws(Exception::class)
+    fun findClosestLocations(latitude: Double, longitude: Double, path: String): List<Shop> {
+
+        val shops = CSVReader.parseCSV(path)
+                ?: throw Exception("Input file can not be parsed.")
+
+        if (shops.size == 0) {
+            throw Exception("Invalid input file.")
+        }
+
+        if (shops.size == 1) {
+            return shops
+        } else {
+            val userLocation = Location("userLocation")
+
+            userLocation.latitude = latitude
+            userLocation.longitude = longitude
+
+            for (shop in shops) {
+                shop.distance = calculateDistance(userLocation, shop.location)
+            }
+
+            val sortedshops = shops.sortedWith(compareBy { it.distance })
+
+            printForTest(getfirstMax3Elements(sortedshops))
+
+            return getfirstMax3Elements(sortedshops)
+        }
+    }
+
+    fun calculateDistance(src: Location, dest: Location?): Float{
+        return src.distanceTo(dest)
+    }
+
+    fun getfirstMax3Elements(list: List<Shop>): ArrayList<Shop> {
```

Comment:
There are [easier ways](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/take.html) to do this is kotin

Reply:
I didn`t know about this. Thank you.